common/src/api/external/mod.rs

```diff
@@ -3173,6 +3173,9 @@ mod test {
         for name in valid_names {
             eprintln!("check name \"{}\" (should be valid)", name);
             assert_eq!(name, name.parse::<Name>().unwrap().as_str());
+            // make sure valid names are also valid hostnames so we can use names as
+            // fallback hostnames, e.g., in instance create
+            assert_eq!(name, name.parse::<Hostname>().unwrap().as_str());
```

Comment:
There aren't that many examples in the list. It would be cool to do a proptest type thing here but I didn't see any tests like that in omicron.